### ISSUE TITLE
No for-loop not show up in specific case

### ISSUE THREAD

**Issue by thinh105** [sindresorhus/eslint-plugin-unicorn#806]
In this case, this rule not show up and need to fix:

```js
  const TourPromises = [];
  for (let i = 0; i < tours.length; i += 1) {
    const tour = tours[Number.parseInt(i, 10)]; // not show up
 
//   const tour = tours[i]; // will show up

    TourPromises.push(
      Tour.create(tour).catch((error) => {
        errorTours.push({ tour, error });
      })
    );
  }
```

Also, In the rule "No For-loop", it is recommended to use for-of if possible. 

```js
  for (const [i, element] of abc.entries()) {
    console.log(i);
    element;
  }
```

but it is conflicted with [AirBnb "Using 'ForOfStatement' is not allowed (no-restricted-syntax)"](https://github.com/airbnb/javascript/issues/1271)

```js
iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.eslintno-restricted-syntax
```

So which one should I follow?
Thanks

ps: I just turn the Airbnb rule off!

**Comment by thinh105** [comment]
I just [turn the Airbnb rule Off](https://github.com/airbnb/javascript/issues/1271#issuecomment-548688952)

```json
    "no-restricted-syntax": [
      "error",
      {
        "selector": "ForInStatement",
        "message": "for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries }, and iterate over the resulting array."
      },
      {
        "selector": "LabeledStatement",
        "message": "Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand."
      },
      {
        "selector": "WithStatement",
        "message": "`with` is disallowed in strict mode because it makes code impossible to predict and optimize."
      }
    ],
```

**Comment by sindresorhus** [comment]
> const tour = tours[Number.parseInt(i, 10)]; // not show up

This code makes no sense. Is this the actual code?

> but it is conflicted with AirBnb "Using 'ForOfStatement' is not allowed (no-restricted-syntax)"

Rules from other configs and plugins can conflict. There's nothing we can or want to do about that.
